// flowbite-react/dist/theme/index.js
import { accordionTheme } from '../components/Accordion/theme.js';
import { alertTheme } from '../components/Alert/theme.js';
import { avatarTheme } from '../components/Avatar/theme.js';
import { badgeTheme } from '../components/Badge/theme.js';
import { blockquoteTheme } from '../components/Blockquote/theme.js';
import { breadcrumbTheme } from '../components/Breadcrumb/theme.js';
import { buttonGroupTheme, buttonTheme } from '../components/Button/theme.js';
import { cardTheme } from '../components/Card/theme.js';
import { carouselTheme } from '../components/Carousel/theme.js';
import { checkboxTheme } from '../components/Checkbox/theme.js';
import { clipboardTheme } from '../components/Clipboard/theme.js';
import { darkThemeToggleTheme } from '../components/DarkThemeToggle/theme.js';
import { datePickerTheme } from '../components/Datepicker/theme.js';
import { drawerTheme } from '../components/Drawer/theme.js';
import { dropdownTheme } from '../components/Dropdown/theme.js';
import { fileInputTheme } from '../components/FileInput/theme.js';
import { floatingLabelTheme } from '../components/FloatingLabel/theme.js';
import { footerTheme } from '../components/Footer/theme.js';
import { helperTextTheme } from '../components/HelperText/theme.js';
import { hrTheme } from '../components/HR/theme.js';
import { kbdTheme } from '../components/Kbd/theme.js';
import { labelTheme } from '../components/Label/theme.js';
import { listTheme } from '../components/List/theme.js';
import { listGroupTheme } from '../components/ListGroup/theme.js';
import { megaMenuTheme } from '../components/MegaMenu/theme.js';
import { modalTheme } from '../components/Modal/theme.js';
import { navbarTheme } from '../components/Navbar/theme.js';
import { paginationTheme } from '../components/Pagination/theme.js';
import { popoverTheme } from '../components/Popover/theme.js';
import { progressTheme } from '../components/Progress/theme.js';
import { radioTheme } from '../components/Radio/theme.js';
import { rangeSliderTheme } from '../components/RangeSlider/theme.js';
import { ratingAdvancedTheme, ratingTheme } from '../components/Rating/theme.js';
import { selectTheme } from '../components/Select/theme.js';
import { sidebarTheme } from '../components/Sidebar/theme.js';
import { spinnerTheme } from '../components/Spinner/theme.js';
import { tableTheme } from '../components/Table/theme.js';
import { tabsTheme } from '../components/Tabs/theme.js';
import { textareaTheme } from '../components/Textarea/theme.js';
import { textInputTheme } from '../components/TextInput/theme.js';
import { timelineTheme } from '../components/Timeline/theme.js';
import { toastTheme } from '../components/Toast/theme.js';
import { toggleSwitchTheme } from '../components/ToggleSwitch/theme.js';
import { tooltipTheme } from '../components/Tooltip/theme.js';
export { ThemeConfig } from './config.js';
export { ThemeModeScript, getThemeModeScript, initThemeMode } from './mode-script.js';
export { ThemeProvider, useThemeProvider } from './provider.js';

const theme = {
  accordion: accordionTheme,
  alert: alertTheme,
  avatar: avatarTheme,
  badge: badgeTheme,
  blockquote: blockquoteTheme,
  breadcrumb: breadcrumbTheme,
  button: buttonTheme,
  buttonGroup: buttonGroupTheme,
  card: cardTheme,
  carousel: carouselTheme,
  checkbox: checkboxTheme,
  clipboard: clipboardTheme,
  darkThemeToggle: darkThemeToggleTheme,
  datepicker: datePickerTheme,
  drawer: drawerTheme,
  dropdown: dropdownTheme,
  fileInput: fileInputTheme,
  floatingLabel: floatingLabelTheme,
  footer: footerTheme,
  helperText: helperTextTheme,
  hr: hrTheme,
  kbd: kbdTheme,
  label: labelTheme,
  list: listTheme,
  listGroup: listGroupTheme,
  megaMenu: megaMenuTheme,
  modal: modalTheme,
  navbar: navbarTheme,
  pagination: paginationTheme,
  popover: popoverTheme,
  progress: progressTheme,
  radio: radioTheme,
  rangeSlider: rangeSliderTheme,
  rating: ratingTheme,
  ratingAdvanced: ratingAdvancedTheme,
  select: selectTheme,
  sidebar: sidebarTheme,
  spinner: spinnerTheme,
  table: tableTheme,
  tabs: tabsTheme,
  textarea: textareaTheme,
  textInput: textInputTheme,
  timeline: timelineTheme,
  toast: toastTheme,
  toggleSwitch: toggleSwitchTheme,
  tooltip: tooltipTheme
};

export { theme };
