// flowbite-react/dist/cli/consts.js
import path__default from 'path';

const pluginName = "flowbiteReact";
const pluginPath = "flowbite-react/plugin";
const classListFile = "class-list.json";
const configFile = "config.json";
const outputDir = ".flowbite-react";
const packageJsonFile = "package.json";
const processIdFile = "pid";
const vscodeDir = ".vscode";
const classListFilePath = path__default.join(outputDir, classListFile);
const configFilePath = path__default.join(outputDir, configFile);
const excludeDirs = [
  ".astro",
  ".contentlayer",
  ".git",
  ".next",
  ".parcel-cache",
  ".turbo",
  ".vercel",
  ".vscode",
  "build",
  "coverage",
  "dist",
  "node_modules",
  "out",
  "storybook-static"
];
const allowedExtensions = [".astro", ".js", ".jsx", ".md", ".mdx", ".ts", ".tsx"];
const automaticClassGenerationMessage = `Components specified in ${configFilePath}. Automatic class generation is disabled.`;

export { allowedExtensions, automaticClassGenerationMessage, classListFile, classListFilePath, configFile, configFilePath, excludeDirs, outputDir, packageJsonFile, pluginName, pluginPath, processIdFile, vscodeDir };
